NDTensors/Project.toml

```diff
@@ -36,4 +37,28 @@ StaticArrays = "0.12, 1.0"
 Strided = "0.3, 1"
 TimerOutputs = "0.5.5"
 TupleTools = "1.2.0"
+Preferences = "1.3.0"
+ZygoteRules = "0.2.2"
```

Comment:
Are these needed as direct dependencies?

Reply:
Oh I see, I thought because it was in `[compat]` it meant that it was a direct dependency, but you are using `[compat]` to set the compatible versions of the `[extras]` as well. But I don't see that you are setting compat entries for the other `[extras]`, is this one in particular needed?

Reply:
I was just matching the compat versions from `ITensors`.

Reply:
In general, I haven't been putting `[compat]` entries for test dependencies unless it is necessary, like if our tests fail with a newer version of a package and we want to specify the last version where the tests work. In those cases where we do want to specify the `[compat]` of a test dependency, I think the right place to put it would be in `NDTensors/test/Project.toml`, not in `NDTensors/Project.toml`.